ObjectOrientedProgramming/MemoryOwnership/RAIICopySwapIdiom/src/dynamic_array.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CPPBASICS_DYNAMIC_ARRAY_H
+#define CPPBASICS_DYNAMIC_ARRAY_H
+
+#include <iostream>
+
+class dynamic_array {
```

Comment:
Maybe would make sense to rename the class to `dynarray`?
`dynarray` was proposed for addition into C++ standard library some time ago, but the idea was abandoned. 
https://stackoverflow.com/questions/20777623/what-is-the-status-on-dynarrays/

Still `dynarray` name is somewhat famous in C++ world and the thing is often implemented in various custom libraries.